### ISSUE TITLE
Provide parent as DOM node instead of selector?

### ISSUE THREAD

**Issue by billyjanitsch** [rstacruz/nprogress#124]
Currently, the `parent` config option takes a string query selector. It would be great if the option could alternatively take an actual DOM node. This would let NProgress integrate more elegantly with view frameworks like Backbone/Marionette where DOM elements of interest are already cached and don't otherwise need a unique selector.

**Comment by chrisdrackett** [comment]
agreed, its actually a pain for us to have to add a query selector as we already have the element we need.

**Comment by sly7-7** [comment]
I think it used to work when nprogress was depending on jquery. 
For now, as a workaround maybe you can generate an uuid when it does not already exist on the node.

**Comment by rstacruz** [comment]
:+1: pr welcome for this.

**Comment by andrewbranch** [comment]
I'm working on a PR for this. Will submit today.
